sdc/extensions/indexes/int64_index_ext.py

```diff
@@ -0,0 +1,413 @@
+    if not isinstance(self, Int64IndexType):
+        return None
+
+    def pd_int64_index_values_impl(self):
+        return self._data
+
+    return pd_int64_index_values_impl
+
+
+@sdc_overload(len)
+def pd_int64_index_len_overload(self):
+    if not isinstance(self, Int64IndexType):
+        return None
+
+    def pd_int64_index_len_impl(self):
+        return len(self._data)
+
+    return pd_int64_index_len_impl
+
+
+@sdc_overload(operator.contains)
+def pd_int64_index_contains_overload(self, val):
+    if not isinstance(self, Int64IndexType):
+        return None
+
+    _func_name = 'Operator contains().'
+    ty_checker = TypeChecker(_func_name)
+
+    if not (isinstance(val, types.Integer)):
+        ty_checker.raise_exc(val, 'integer scalar', 'val')
+
+    def pd_int64_index_contains_impl(self, val):
+        # TO-DO: add operator.contains support for arrays in Numba
+        for i in prange(len(self._data)):
+            if val == self._data[i]:
+                break
+        else:
+            return False
+
+        return True
+
+    return pd_int64_index_contains_impl
+
+
+@sdc_overload_method(Int64IndexType, 'copy')
+def pd_int64_index_copy_overload(self, name=None, deep=False, dtype=None):
+    if not isinstance(self, Int64IndexType):
+        return None
+
+    _func_name = 'Method copy().'
+    ty_checker = TypeChecker(_func_name)
+
+    if not (isinstance(name, (types.NoneType, types.Omitted, types.UnicodeType)) or name is None):
+        ty_checker.raise_exc(name, 'string or none', 'name')
+
+    if not (isinstance(deep, (types.Omitted, types.Boolean)) or deep is False):
+        ty_checker.raise_exc(deep, 'boolean', 'deep')
+
+    if not _check_dtype_param_type(dtype):
+        ty_checker.raise_exc(dtype, 'int64 dtype', 'dtype')
+
+    name_is_none = isinstance(name, (types.NoneType, types.Omitted)) or name is None
+    keep_name = name_is_none and self.is_named
+
+    def pd_int64_index_copy_impl(self, name=None, deep=False, dtype=None):
+
+        _name = self._name if keep_name == True else name  # noqa
+        return init_int64_index(self._data, _name)
+
+    return pd_int64_index_copy_impl
+
+
+@sdc_overload(operator.getitem)
+def pd_int64_index_getitem_overload(self, idx):
+    if not isinstance(self, Int64IndexType):
+        return None
+
+    _func_name = 'Operator getitem().'
+    ty_checker = TypeChecker(_func_name)
+
+    if not (isinstance(idx, (types.Integer, types.SliceType))
+            or isinstance(idx, (types.Array, types.List)) and isinstance(idx.dtype, (types.Integer, types.Boolean))):
+        ty_checker.raise_exc(idx, 'integer, slice, integer array or list', 'idx')
+
+    if isinstance(idx, types.Integer):
+        def pd_int64_index_getitem_impl(self, idx):
+            index_len = len(self._data)
+            # FIXME_Numba#5801: Numba type unification rules make this float
+            idx = types.int64((index_len + idx) if idx < 0 else idx)
+            if (idx < 0 or idx >= index_len):
```

Comment:
That's true, but I'm not sure compiler can remove dead branch from the loop body basing on loop variable range (or am I wrong?) It may be so that we would need checked/unchecked impl (same as Numba does for lists). Need to dig this further.